pkg/core/terraform/tf.go

```diff
@@ -155,33 +157,70 @@ func (sw *StdWriter) GetString() string {
 	return s
 }
 
-func (tf Terraform) formatTerraformWorkspaces(list string) (string) {
+func (tf Terraform) formatTerraformWorkspaces(list string) string {
 
 	list = strings.TrimSpace(list)
 	char_replace := strings.NewReplacer("*", "", "\n", ",", " ", "")
 	list = char_replace.Replace(list)
 	return list
 }
 
+func copyGitHubVars() (map[string]string, error) {
+	githubVars := os.Getenv("GITHUB_VARS")
+	var result = make(map[string]string)
+	if githubVars != "" {
+		parsedJson := make(map[string]string)
+		err := json.Unmarshal([]byte(githubVars), &parsedJson)
+		if err != nil {
+			return nil, err
+		}
+
+		for k, v := range parsedJson {
+			err := os.Setenv(k, v)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return result, nil
+}
+
 func (tf Terraform) Plan(params []string, envs map[string]string) (bool, string, string, error) {
-	workspaces, _, _, err := tf.runTerraformCommand("workspace", envs, "list")
+	githubVars, _ := copyGitHubVars()
+	// merge envs and github vars maps
+	if githubVars != nil {
+		for k, v := range githubVars {
+			envs[k] = v
+		}
+	}
+
+	expandedParams := make([]string, 0)
+	for _, p := range params {
+		s := os.ExpandEnv(p)
+		s = strings.TrimSpace(s)
+		if s != "" {
+			expandedParams = append(expandedParams, s)
+		}
+	}
+
+	workspaces, _, _, err := tf.runTerraformCommand("workspace", nil, "list")
 	if err != nil {
 		return false, "", "", err
 	}
 	workspaces = tf.formatTerraformWorkspaces(workspaces)
 	if strings.Contains(workspaces, tf.Workspace) {
-		_, _, _, err := tf.runTerraformCommand("workspace", envs, "select", tf.Workspace)
+		_, _, _, err := tf.runTerraformCommand("workspace", nil, "select", tf.Workspace)
```

Comment:
I would check that you need to do this again

Reply:
it seems it can be reverted back, it didn't work when I was mapping envs to vars, but I removed that code so it should be ok